workers/unity/Assets/Fps/Scripts/GameLogic/Animation/FpsAnimator.cs

```diff
@@ -195,33 +197,59 @@ private static float CalculateAngle(Transform from, Vector3 direction)
 
         private void SetBoolParameter(AnimationBoolParameter parameter, bool newValue)
         {
-            if (parameter.LastValue != newValue)
+            if (parameter.LastValue == newValue)
             {
-                parameter.LastValue = newValue;
-                foreach (var animator in animators)
-                {
-                    animator.SetBool(parameter.Name, newValue);
-                }
+                return;
+            }
+
+            parameter.LastValue = newValue;
+            foreach (var animator in animators)
+            {
+                animator.SetBool(parameter.Name, newValue);
             }
         }
 
         private void Update()
         {
-            LerpFloatParameter(pitchParameter, animationSettings.InterpolatePitchSpeed);
-            LerpFloatParameter(xParameter, animationSettings.InterpolateMovementSpeed);
-            LerpFloatParameter(zParameter, animationSettings.InterpolateMovementSpeed);
-            LerpFloatParameter(movementParameter, animationSettings.InterpolateMovementSpeed);
+            LerpFloatParameter(pitchParameter, animationSettings.InterpolatePitchDuration);
+            LerpFloatParameter(xParameter, animationSettings.InterpolateMovementDuration);
+            LerpFloatParameter(zParameter, animationSettings.InterpolateMovementDuration);
+            LerpFloatParameter(movementParameter, animationSettings.InterpolateMovementDuration);
+        }
+
+        private static void SetFloatParameter(AnimationFloatParameter parameter, float newValue)
```

Comment:
Make this a setter on the `AnimationFloatParameter.TargetValue` property so we can get rid of this function

Reply:
Ah yes, will do